packages/xsnap/test/xsnap-eof.test.js

```diff
@@ -0,0 +1,268 @@
+  });
+
+  p.on('error', err => {
+    nodeError = err;
+  });
+
+  p.on('exit', (code, signal) => {
+    nodeExited = true;
+    nodeExitCode = code;
+    nodeExitSignal = signal;
+  });
+
+  p.stdin.end();
+
+  await waitFor(() => {
+    return nodeExited;
+  });
+
+  t.is(`${strOut}`, ``, 'wrong stdout output');
+  t.regex(`${strErr}`, /Has parent died\?/, 'wrong stderr output');
+  t.is(nodeError, null, 'parent returned an error');
+  t.is(nodeExited, true, 'parent did not exit');
+  t.is(nodeExitCode, null, 'parent exit code is not null');
+  t.is(`${nodeExitSignal}`, `SIGKILL`, 'wrong parent exit signal');
+});
+
+async function doSetup(handleCommand) {
+  let xsnapProcess;
+  const spawnSpy = (...args) => {
+    xsnapProcess = proc.spawn(...args);
+    return xsnapProcess;
+  };
+
+  const io = { spawn: spawnSpy, os: os.type(), fs, tmpName };
+  const vat = await xsnap({ ...options(io), handleCommand, stderr: 'pipe' });
+  await vat.evaluate(
+    'function handleCommand(message) { issueCommand(new Uint8Array().buffer); };',
+  );
+  return { vat, xsnapProcess };
+}
+
+test('xsnap-worker complains while trying to READ an answer when pipes are closed', async t => {
+  let xsnapProcess;
+  let toXsnap;
+  let fromXsnap;
+  let issueCommandError;
+  let vatCloseError;
+  let vatExitCode;
+  let vatExitSignal;
+  let strErr = '';
+
+  async function handleCommand(_message) {
+    // suspend the child process so the pipes appear
+    // to close atomically from its perspective
+    xsnapProcess.kill('SIGSTOP');
+    // wait for signal delivery
+    await waitFor(() => {
+      return xsnapProcess.killed;
+    });
+
+    // close the pipes
+    toXsnap.end();
+    toXsnap.destroy();
+    fromXsnap.end();
+    fromXsnap.destroy();
+
+    // un-suspend the child process
+    xsnapProcess.kill('SIGCONT');
+    // wait for signal delivery
+    await waitFor(() => {
+      return xsnapProcess.killed;
+    });
+
+    return new Uint8Array();
+  }
+
+  const setup = await doSetup(handleCommand);
+  const { vat } = setup;
+  ({ xsnapProcess } = setup);
+
+  toXsnap = xsnapProcess.stdio[3];
+  fromXsnap = xsnapProcess.stdio[4];
+
+  xsnapProcess.on('exit', (code, signal) => {
+    vatExitCode = code;
+    vatExitSignal = signal;
+  });
+
+  xsnapProcess.stdio[2].on('data', data => {
+    strErr = `${strErr}${data}`;
+  });
+
+  try {
+    // this will error out since we are intentionally
+    // breaking communications with the worker
+    await vat.issueStringCommand('0');
+  } catch (err) {
+    issueCommandError = err;
+  }
+
+  // wait for the child to exit
+  await waitFor(() => {
+    return vatExitCode || vatExitSignal;
+  });
+
+  try {
+    // this will error out since vat has died.
+    await vat.close();
+  } catch (err) {
+    vatCloseError = err;
+  }
+  t.truthy(issueCommandError, 'issueCommand() did not error out');
+  t.truthy(vatCloseError, 'vat.close() did not error out');
+  t.falsy(vatExitSignal, 'wrong vat exit signal');
```

Comment:
`is` and `not` give better output on failure.
```suggestion
  t.not(issueCommandError, undefined, 'issueCommand() did not error out');
  t.not(vatCloseError, undefined, 'vat.close() did not error out');
  t.is(vatExitSignal, null, 'wrong vat exit signal');
```